.evergreen/start-orchestration.sh

```diff
@@ -34,8 +34,8 @@ echo "Creating virtual environment 'venv'..."
 venvcreate "${PYTHON:?}" venv
 echo "Creating virtual environment 'venv'... done."
 
-# Install from github to get the latest mongo-orchestration.
-python -m pip install -q --upgrade 'https://github.com/mongodb/mongo-orchestration/archive/master.tar.gz'
+# Install from github to get the latest mongo-orchestration, fall back on published wheel.
+python -m pip install -q --upgrade 'https://github.com/mongodb/mongo-orchestration/archive/master.tar.gz' || python -m pip install -q --upgrade mongo-orchestration
```

Comment:
We intentionally install from master, why fallback to the published one?

Reply:
On rhel 7.0 (but not rhel 7.9), we cannot install hatchling because of the version of glibc.

Reply:
Oh that's lame, could you add a comment explaining that?

Reply:
Wait why does hatchling install fail? It looks like a pure python library: https://pypi.org/project/hatchling/#files

Reply:
Ah, the real issue is that find_python3 is using Python 3.6, for which there is no matching version of hatchling.  I'll update the comment, because we could still end up in a whack-a-mole situation.

Reply:
Here's the failing build: https://spruce.mongodb.com/task/drivers_tools_tests_os_requires_50__os_requires_50~rhel70_small_auth~auth_ssl~ssl_test_5.0_standalone_patch_cf39786af0437b52a0f9829190d55b11bfbfe354_66d8ab2dfbe79f0007abf9fb_24_09_04_18_47_19/logs?execution=0

Reply:
Ah so the real issue is that we're using Python 3.6 but mongo-orc@master requires Python 3.8+